sql/core/src/main/scala/org/apache/spark/sql/Column.scala

```diff
@@ -37,6 +39,16 @@ private[sql] object Column {
   def apply(expr: Expression): Column = new Column(expr)
 
   def unapply(col: Column): Option[Expression] = Some(col.expr)
+
+  @scala.annotation.varargs
+  def updateExpressionsOrigin(cols: Column*): Unit = {
+    // Update Expression.origin using the callSite of an operation
+    val callSite = org.apache.spark.util.Utils.getCallSite().shortForm
+    cols.map(col => col.expr.foreach(e => e.origin.callSite = Some(callSite)))
+    // Update CurrentOrigin for setting origin for LogicalPlan node
+    CurrentOrigin.set(
+      Origin(Some(callSite), CurrentOrigin.get.line, CurrentOrigin.get.startPosition))
```

Comment:
Maybe, we can use `withOrigin` to set CurrentOrigin temporarily.
Also, we may need to create CurrentOrigin for each expression because in `TreeNode`, `origin` is set by `CurrentOrigin.get`.
Otherwise, some non-related expressions share a same origin.